lsp/glint.lua

```diff
@@ -22,27 +22,37 @@
 ---   },
 --- })
 
+function get_cmd()
+  local useGlobal = vim.lsp.config.glint.init_options.glint.useGlobal
+  if useGlobal then
+    return { 'glint-language-server' }
+  end
+
+  local root_markers = vim.lsp.config.glint.root_markers
+  if not root_markers or #root_markers == 0 then
+    vim.notify('No root markers defined for glint', vim.log.levels.ERROR)
+    return nil
+  end
+
+  local root_dir = vim.fs.dirname(vim.fs.find(root_markers, { upward = true })[1])
+  if not root_dir then
+    vim.notify('Could not find root directory for glint', vim.log.levels.ERROR)
+    return nil
+  end
```

Comment:
this is kind of strange. `cmd()` is not going to be invoked unless the config was activated. Nvim already checks `root_markers` and `root_dir` to decide whether to activate a client for a given buffer.

when would this not be the case ? we should avoid reinventing the vim.lsp framework here.

Reply:
oke so then I guess the checks are not needed.

Is there a way for me to access the client `root_dir` from within `get_cmd()`?
I could not find in in `vim.lsp`. 

Or would you recommend using lspconfig.util `root_pattern`?

I would prefer not having to find it again myself.